lint/importShadow_checker.go

```diff
@@ -42,18 +42,27 @@ type importShadowChecker struct {
 }
 
 func (c *importShadowChecker) VisitLocalDef(def astwalk.Name, _ ast.Expr) {
-	for _, imp := range c.ctx.Context.pkg.Imports() {
-		if imp.Name() == def.ID.Name {
-			c.warnImportShadowed(def.ID, imp)
+	for _, v := range c.ctx.Context.typesInfo.Defs {
+		if v == nil || v.Parent() == nil {
+			continue
+		}
+		for _, name := range v.Parent().Names() {
```

Comment:
AFAIK `Names()` call does build slice for every call. It also does sorting.
Not very good API for this use case. :(

Reply:
yes it is... it is used here because (*types.Scope).elems is not exported and only possible solution to iterate over it is to use (*types.Scope).Lookup(name string), let me remove this iteration completely so it become a little bit faster.

Reply:
done